bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/InstallationCleaner.java

```diff
@@ -153,7 +159,9 @@ private void newProcess() throws IOException {
                 .redirectError(ProcessBuilder.Redirect.INHERIT)
                 .redirectOutput(ProcessBuilder.Redirect.INHERIT)
                 .directory(new File(System.getProperty("user.dir")));
-        builder.start();
+        process = builder.start();
+        process.waitFor(environment.getTimeout(), TimeUnit.SECONDS);
```

Comment:
This also sounds a bit inappropriate, since this method could be invoked directly from the shutdown hook thread. The shutdown hook API says that is inadvisable to attempt any user interaction or to perform a long-running computation in a shutdown hook.

I guess my question would be if this is somehow killed completely by the JVM, if there could be possibilities of having the started process running around.

In any case, if decided, we can move on and see how it behaves in the CI

Reply:
@yersan , I was thinking to it more, and I think that we shouldn't merge it. Although I am confident on the Linux fix, it requires more on Windows front.